src/org/loklak/api/admin/StatusService.java

```diff
@@ -74,34 +74,8 @@ public static JSONObject status(final String protocolhostportstub) throws IOExce
         return json;
     }
 
-    @Override
-    public JSONObject serviceImpl(Query post, HttpServletResponse response, Authorization rights, JSONObjectWithDefault permissions) throws APIException {
-
-        if (post.isLocalhostAccess() && OS.canExecUnix && post.get("upgrade", "").equals("true")) {
-            Caretaker.upgrade(); // it's a hack to add this here, this may disappear anytime
-        }
-
-        final String backend = DAO.getConfig("backend", "");
-        final boolean backend_push = DAO.getConfig("backend.push.enabled", false);
-        JSONObject backend_status = null;
-        JSONObject backend_status_index_sizes = null;
-        if (backend.length() > 0 && !backend_push) {
-            try {
-                backend_status = StatusService.status(backend);
-                backend_status_index_sizes = backend_status == null ? null : (JSONObject) backend_status.get("index_sizes");
-            } catch (IOException e) {}
-        }
-        long backend_messages = backend_status_index_sizes == null ? 0 : ((Number) backend_status_index_sizes.get("messages")).longValue();
-        long backend_users = backend_status_index_sizes == null ? 0 : ((Number) backend_status_index_sizes.get("users")).longValue();
-        long local_messages = DAO.countLocalMessages();
-        long local_users = DAO.countLocalUsers();
-
-        post.setResponse(response, "application/javascript");
-
-        // generate json
-        Runtime runtime = Runtime.getRuntime();
-        JSONObject json = new JSONObject(true);
-        JSONObject system = new JSONObject(true);
+    public static JSONObject system(Runtime runtime) throws Exception {
```

Comment:
Method name can be more clear, e.g. `getSystemConfig`. Also for other methods in the diff.